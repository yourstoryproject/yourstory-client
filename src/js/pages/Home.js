import React from 'react';
import Stories from '../components/Stories';

export default class Home extends React.Component {
	render() {
		return (
			<div className="home-container">
				<Stories />
			</div>
		);
	}
}
