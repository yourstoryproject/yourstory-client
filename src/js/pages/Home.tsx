import EmailAggregationForm from 'components/EmailAggregationForm';
import React from 'react';
import logo from 'resources/images/logo-banner-768.png';
import styled from 'styled-components';

const HomeContainer = styled.div`
	display: flex;
	flex-direction: column;
	padding: 16px;
`;

const LogoBanner = styled.img`
	margin: 24px auto;
	width: 90%;

	@media (min-width: 425px) {
		width: 70%;
	}

	@media (min-width: 768px) {
		width: 60%;
	}
`;

const Home = () => {
	return (
		<HomeContainer>
			<LogoBanner alt="Your Story Project Logo" src={logo} />
			<EmailAggregationForm />
		</HomeContainer>
	);
};

export default Home;
