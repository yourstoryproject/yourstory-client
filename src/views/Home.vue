<template>
  <div class="home">
    <img class="logo" alt="Your Story Project logo" src="../assets/logo.png" />
    <TheHomePageHeader />
    <img
      class="world-image"
      alt="Picture of earth with lines"
      src="../assets/world-image.png"
    />
    <LargeButton
      superLabel="Want to be a part of the experiment?"
      label="Start a Notebook"
      :callback="navigateStartANotebook"
    />
  </div>
</template>

<script>
import TheHomePageHeader from '@/components/TheHomePageHeader';
import LargeButton from '@/components/LargeButton';

export default {
  name: 'Home',
  components: { TheHomePageHeader, LargeButton },
  methods: {
    navigateStartANotebook() {
      // Anti-patter: Buttons should not navigate.
      this.$router.push({ path: 'start-a-notebook' });
    },
  },
};
</script>

<style lang="scss" scoped>
.logo {
  height: 100px;
}

.world-image {
  width: 100%;
  margin-top: 24px;
  margin-bottom: 24px;
}
</style>
