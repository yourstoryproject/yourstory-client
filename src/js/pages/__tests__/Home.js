import React from 'react';
import Home from '../Home';
import { shallow } from 'enzyme';

describe('Home', () => {
	it('should render', () => {
		const component = shallow(<Home />);
		expect(component).toMatchSnapshot();
	});
});
