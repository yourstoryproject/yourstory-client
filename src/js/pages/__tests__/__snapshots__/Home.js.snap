// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render 1`] = `
<styled.div>
  <styled.img
    alt="Your Story Project Logo"
    src={
      Object {
        "process": [Function],
      }
    }
  />
  <EmailAggregationForm />
</styled.div>
`;
